plugin/src/main/java/io/jenkins/plugins/casc/SchemaGeneration.java

```diff
@@ -0,0 +1,156 @@
+package io.jenkins.plugins.casc;
+
+import java.util.*;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator;
+
+public class SchemaGeneration {
+
+    public static String generateSchema() {
+
+        /**
+         * The initial template for the JSON Schema
+         */
+        StringBuilder schemaString = new StringBuilder();
```

Comment:
I think with we switch to using JSON builder seems tedious to use strings.

Reply:
JSON builder might refuse the trailing commas

Reply:
When using a JSON builder you should not be concerned about the delimiters you should be using JSON objects and JSON arrays to build out the schema.
Feels you from thinking how to write json in a string format.